src/Runner.Worker/ExecutionContext.cs

```diff
@@ -411,21 +432,29 @@ public TaskResult Complete(TaskResult? result = null, string currentOperation =
                 this.Warning($"The job has experienced {TimeSpan.FromMilliseconds(_totalThrottlingDelayInMilliseconds).TotalSeconds} seconds total delay caused by server throttling.");
             }
 
+            DateTime now = DateTime.UtcNow;
             _record.CurrentOperation = currentOperation ?? _record.CurrentOperation;
             _record.ResultCode = resultCode ?? _record.ResultCode;
-            _record.FinishTime = DateTime.UtcNow;
+            _record.FinishTime = now;
             _record.PercentComplete = 100;
             _record.Result = _record.Result ?? TaskResult.Succeeded;
             _record.State = TimelineRecordState.Completed;
 
+            foreach (var issue in _embeddedIssueCollector)
+            {
+                // Before our main timeline's final QueueTimelineRecordUpdate,
+                //    inject any issues collected by embedded ExecutionContexts.
+                AddIssue(issue);
```

Comment:
Is there a better point in time to do this work?  This was the best spot I could find -- right after other (more fundamental) `_record` updates and (more importantly) before the final call to `QueueTimelineRecordUpdate`

Reply:
I'm not sure we care about the fundamental record updates on _record for composite steps (embedded).

for embedded steps it may just be okay to mask secrets and output to logs, and when we bubble fully up to the parent, then we can update only the parent's timeline record.